cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java

```diff
@@ -212,20 +213,29 @@ private static Path validateBala(BuildProject project, CentralAPIClient client)
     private static Path validateBalaFile(BuildProject project) {
         final PackageName pkgName = project.currentPackage().packageName();
         final PackageOrg orgName = project.currentPackage().packageOrg();
+        PackageVersion packageVersion = project.currentPackage().packageVersion();
 
         // Get bala output path
         Path balaOutputDir = project.currentPackage().project().sourceRoot().resolve(ProjectConstants.TARGET_DIR_NAME)
                 .resolve(ProjectConstants.TARGET_BALA_DIR_NAME);
 
         if (Files.notExists(balaOutputDir)) {
             throw new ProjectException("cannot find bala file for the package: " + pkgName + ". Run "
-                    + "'bal build' to compile and generate the bala.");
+                    + "'bal build -c' to compile and generate the bala.");
         }
 
         Path packageBalaFile = findBalaFile(pkgName, orgName, balaOutputDir);
         if (null == packageBalaFile) {
             throw new ProjectException("cannot find bala file for the package: " + pkgName + ". Run "
-                    + "'bal build' to compile and generate the bala.");
+                    + "'bal build -c' to compile and generate the bala.");
+        }
+
+        if (!packageBalaFile.toString().endsWith(
```

Comment:
Hi, this is another validation we added to confirm with the Ballerina.toml version and We still send the version taken from bala right?

Reply:
Approved the PR after checking this.

Reply:
We don't continue pushing if the validation fails. The versions should be identical to be able to push.